src/ExactDiagonalization/exact_diagonalization_problem.jl

```diff
@@ -0,0 +1,143 @@
+"""
+    ExactDiagonalizationProblem(hamiltonian::AbstractHamiltonian, [v0]; kwargs...)
+
+Defines an exact diagonalization problem with an [`AbstractHamiltonian`](@ref) `hamiltonian`.
+Optionally, a starting vector of type [`AbstractDVec`](@ref), or a single address or a
+collection of addresses can be passed as `v0`.
+
+`ExactDiagonalizationProblem`s can be solved with
+[`solve`](@ref solve(::ExactDiagonalizationProblem)).
+
+# Keyword arguments
+- `algorithm=LinearAlgebraSolver()`: The algorithm to use for solving the problem. The
```

Comment:
Looks like I found a bug. It seems like setting the algorithm in the problem selects the correct solver, but attempts to pass the `algorithm` on to the solver. This errors for KrylovKit and Arpack, but gives a warning with LinearAlgebra.

MWE:

```julia
using Rimu
using KrylovKit

H = HubbardReal1D(BoseFS(1,1,1,1))
prob = ExactDiagonalizationProblem(H; algorithm=KrylovKitSolver())
solve(prob)
# ERROR: MethodError: no method matching eigselector(::KrylovKitExt.OperatorMultiplier{HubbardReal1D{Float64, BoseFS{4, 4, BitString{7, 1, UInt8}}, 1.0, 1.0}, PDWorkingMemory{BoseFS{4, 4, BitString{7, 1, UInt8}}, Float64, Rimu.DictVectors.NonInitiatorValue{Float64}, IsDeterministic{Float64, NoCompression}, NonInitiator, Rimu.DictVectors.NotDistributed, 1}}, ::Type{Any}; ishermitian::Bool, issymmetric::Bool, algorithm::KrylovKitSolver{true})

```

Reply:
Thanks for the bug report. It is now fixed.